src/main/java/org/opentripplanner/routing/core/State.java

```diff
@@ -200,12 +266,30 @@ public long getElapsedTimeSeconds() {
         return Math.abs(getTimeSeconds() - stateData.startTime);
     }
 
+    public boolean isCompatibleBikeRentalState(State state) {
+        return stateData.bikeRentalState ==   state.stateData.bikeRentalState;
+    }
+
+    public boolean isBikeRentingFromStation() {
+        return stateData.bikeRentalState == BikeRentalState.RENTING_FROM_STATION;
+    }
+
     public boolean isBikeRenting() {
-        return stateData.usingRentedBike;
+        return stateData.bikeRentalState == BikeRentalState.RENTING_FROM_STATION
+            || stateData.bikeRentalState == BikeRentalState.RENTING_FLOATING;
+    }
+
+    public boolean bikeRentalIsFinished() {
+        return stateData.bikeRentalState == BikeRentalState.HAVE_RENTED
+            || stateData.bikeRentalState == BikeRentalState.RENTING_FLOATING;
+    }
```

Comment:
This method does not seem to account for the possibility of a vehicle rented from a station that is dropped off in a floating state. It is unclear from the GBFS spec (see https://github.com/NABSA/gbfs/issues/270) whether this is allowed or forbidden by default, but in my personal experience, I believe most systems generally allow floating dropoffs from vehicles picked up at stations if the system allows floating dropoffs in the first place.